p256/src/arithmetic/projective.rs

```diff
@@ -194,16 +193,37 @@ impl ProjectivePoint {
 
     /// Returns `[k] self`.
     fn mul(&self, k: &Scalar) -> ProjectivePoint {
-        let mut ret = ProjectivePoint::IDENTITY;
-
-        for limb in k.limbs().iter().rev() {
-            for i in (0..Limb::BIT_SIZE).rev() {
-                ret = ret.double();
-                ret.conditional_assign(&(ret + self), Choice::from(((limb.0 >> i) & 1) as u8));
+        let mut pc = [ProjectivePoint::default(); 16];
+        pc[0] = ProjectivePoint::IDENTITY;
```

Comment:
FWIW, `ProjectivePoint::default()` returns `ProjectivePoint::IDENTITY`.

I suppose you're initializing it explicitly because `ProjectivePoint::default()` is effectively a placeholder for an "uninitialized" element of the array which will be overwritten?

Reply:
Exactly.

 Elements are initialized right after and the compiler doesn't emit any initialization code.